backend/internal/controllers/utils.go

```diff
@@ -8,5 +8,6 @@ import (
 // Sends data and status code for http
 func writeResponse(rw http.ResponseWriter, statusCode int, body interface{}) {
 	rw.WriteHeader(statusCode)
+	rw.Header().Set("Content-Type", "application/json")
```

Comment:
Actually you need to swap these two lines. `WriteHeader` *literally* writes the entire HTTP header (from `HTTP/<version> <code>` to the end of the headers) to the output stream, so you need to be done with setting headers by then.